core/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsFeedLoader.java

```diff
@@ -0,0 +1,117 @@
+            } catch (ReflectiveOperationException exception) {
+                System.err.println("Possible bug in GTFS annotation processor");
+                exception.printStackTrace();
+                continue;
+            }
+            tableLoaders.put(loader.gtfsFilename(), loader);
+        }
+    }
+
+    private static Reader createFileReader(InputStream stream) throws FileNotFoundException {
+        return new BufferedReader(new InputStreamReader(stream));
+    }
+
+    public String listTableLoaders() {
+        return String.join(" ", tableLoaders.keySet());
+    }
+
+    public void setNumThreads(int numThreads) {
+        this.numThreads = numThreads;
+    }
+
+    public GtfsFeedContainer load(GtfsInput gtfsInput, GtfsFeedName feedName, ValidatorLoader validatorLoader,
+                                  NoticeContainer noticeContainer) throws IOException, InterruptedException {
+        System.out.println("Loading in " + numThreads + " threads");
+        ExecutorService exec = Executors.newFixedThreadPool(numThreads);
+        List<Callable<GtfsTableContainer>> loaderCallables = new ArrayList<>();
+        Map<String, GtfsTableLoader> remainingLoaders = (Map<String, GtfsTableLoader>) tableLoaders.clone();
+        for (String filename : gtfsInput.getFilenames()) {
+            GtfsTableLoader loader = remainingLoaders.remove(filename.toLowerCase());
+            if (loader == null) {
+                noticeContainer.addNotice(new UnexpectedFile(filename));
+            } else {
+                loaderCallables.add(() -> loader.load(
+                        createFileReader(gtfsInput.getFile(filename)), feedName, validatorLoader, noticeContainer));
+            }
+        }
+        ArrayList<GtfsTableContainer> tableContainers = new ArrayList<>();
+        for (GtfsTableLoader loader : remainingLoaders.values()) {
+            tableContainers.add(loader.loadMissingFile(validatorLoader, noticeContainer));
+        }
+        try {
+            try {
+                List<Future<GtfsTableContainer>> results = exec.invokeAll(loaderCallables);
+                tableContainers.ensureCapacity(results.size());
+                results.forEach(f -> {
+                    try {
+                        tableContainers.add(f.get());
+                    } catch (ExecutionException | InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                });
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+            GtfsFeedContainer feed = new GtfsFeedContainer(tableContainers);
+            List<Callable<Void>> validatorCallables = new ArrayList<>();
+            for (FileValidator validator : validatorLoader.createMultiFileValidators(feed)) {
+                validatorCallables.add(() -> {
+                    validator.validate(noticeContainer);
+                    return null;
```

Comment:
The `load()` method does not return null and does not throw exceptions. Do you want me to throw all InterruptedException instead of catching and printing them?

Reply:
Ah, right, sorry, I missed that. No, I think it's fine as-is.